src/packet.rs

```diff
@@ -113,7 +113,23 @@ impl PacketUp {
 
     pub fn is_potential_beacon(&self) -> bool {
         Self::parse_header(self.payload())
-            .map(|header| header.mtype() == lorawan::MType::Proprietary)
+            .map(|header| {
+                header.mtype() == lorawan::MType::Proprietary
+                    && self.payload().len() >= beacon::BEACON_PAYLOAD_SIZE
```

Comment:
`self.payload()` still includes the header byte so it should be `self.payload().len() - 1 >= beacon::BEACON_PAYLOAD_SIZE`

Reply:
I know, but `>=` captures that case and allows for larger beacon payloads :-)

Reply:
True, I was just surprised by the off by one in my test case. It should sufficiently filter proprietary packets this way.

Reply:
Fixed here [e64be24](https://github.com/helium/gateway-rs/pull/437/commits/e64be247d842c242bd7c8d32191e80a8078c916a)